lnrpc/walletrpc/psbt.go

```diff
@@ -0,0 +1,73 @@
+// +build walletrpc
+
+package walletrpc
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil/psbt"
+	"github.com/btcsuite/btcwallet/wtxmgr"
+	"github.com/lightningnetwork/lnd/lnwallet"
+)
+
+const (
+	defaultMinConf = 1
+	defaultMaxConf = math.MaxInt32
+)
+
+// utxoLock is a type that contains an outpoint of an UTXO and its lock lease
+// information.
+type utxoLock struct {
+	lockID     wtxmgr.LockID
+	outpoint   wire.OutPoint
+	expiration time.Time
+}
+
+// verifyInputsUnspent checks that all inputs are contained in the list of
+// known, non-locked UTXOs given.
+func verifyInputsUnspent(inputs []*wire.TxIn, utxos []*lnwallet.Utxo) error {
+	for idx, txIn := range inputs {
+		found := false
+		for _, u := range utxos {
```

Comment:
Likely won't be a big deal unless "massive" scale, but we should leave a TOOD here to use a map to make things a bit more efficient IMO.